Include/Rover/TrialView.hpp

```diff
@@ -241,9 +248,23 @@ namespace Rover {
   template<typename T>
   template<typename Trial>
   TrialView<T>::TrialView(const Trial& t)
-    : m_begin(ConstIterator(t, std::ptrdiff_t(0))),
-      m_end(ConstIterator(t, static_cast<std::ptrdiff_t>(t.size()))),
-      m_size(t.size()) {}
+    : m_size(t.size()),
+      m_get_ptr([begin = t.begin()] (std::ptrdiff_t offset) {
+        return &(*(begin + offset));
+      }),
+      m_begin(m_get_ptr, 0),
```

Comment:
Don't see a benefit to having ```m_begin``` and ```m_end``` as member variables. This will cause issues with the copy constructor. Construct them in their respective methods.